src/Compilers/CSharp/Portable/Symbols/MemberSignatureComparer.cs

```diff
@@ -151,7 +152,7 @@ internal sealed class MemberSignatureComparer : IEqualityComparer<Symbol>
             considerReturnType: true,
             considerTypeConstraints: false,
             considerCallingConvention: false,
-            considerRefKindDifferences: true,
+            refKindCompareMode: RefKindCompareMode.ConsiderDifferences,
```

Comment:
Consider adding a test showing that interceptor matching is strict

Reply:
Adding `InterceptorTests.SignatureMismatch_09`.